packages/changed-elements-react/src/api/VersionCompareTiles.ts

```diff
@@ -173,8 +173,16 @@ export class Provider
   private _eeEmphasized: Id64Set | undefined;
   private _lastJson: EmphasizeElementsProps | undefined;
 
+  /** returns copy of previous ee*/
+  public get lastJson(): EmphasizeElementsProps | undefined {
+    if (this._lastJson) {
+      // return a copy of the last json
+      return JSON.parse(JSON.stringify(this._lastJson));
```

Comment:
There are more efficient ways to clone objects, but even then, unsure if you would need to clone it to begin with?

Reply:
I was trying to prevent modification of the internal state of the class. If I pass back an object, it is a reference, meaning outside consumers can affect the internal state of class-leading to unwanted side effects

Reply:
Also, what would be a better way? I know lodash allows for a deep copy. When should you worry about the case I listed?

Reply:
If your worry is mis-use, could you simply mark the output of this as `Readonly<EmphasizeElementsProps>`?

[See example](https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgCJzHZBvAsAKGSOQBMM4AuZAZzClAHMBuAgXwIIQBs5rrkAKhFo4CxZJFoA1OFCrpMLfGOIIA9iFpQArgjBqoACgBuARnnkANMmMAmC5gCUowuOJgAFsGoA6SWBkoZABeG1slcXZlVyIAB20AIy5gBGQGCADZW0NHKgAlCDgSDS4ATwAeBTgAPhc3YigM7SgQCS9ff0CI4iiozg0RMBDkEAgAd0FhMENsUnIqACIAWVKqheRWa1myTEWVqtt11kclfs0hxpgpU2GwH3TMqGyTgkvrnx2sUIWhWgWmZAAekByAAcmowCkUOoALaxYBcaDIWGxHigahAA)

Clearly someone could typecast it as any, but if the function's signature is marked Readonly, that should suffice to showcase you don't mean it to be overridden. Also, if this data is meant to automatically reset the frontstage when comparison ends, I don't see why you would need to even make it public if its moved to the frontstage manager.